internal/imagemgmt/imagemgmt.go

```diff
@@ -0,0 +1,278 @@
+	// Get the list of containers via crictl
+	output, err := c.Executor.Execute("crictl", "images", "-o", "json")
+	if err != nil {
+		rc = fmt.Errorf("failed to run crictl images: %w", err)
+		return
+	}
+
+	var crictlImages crictlImagesList
+	if err := json.Unmarshal([]byte(output), &crictlImages); err != nil {
+		rc = fmt.Errorf("unable to parse crictl images command output: %w", err)
+		return
+	}
+
+	for _, image := range crictlImages.Images {
+		if image.Pinned {
+			pinned = lcautils.AppendToListIfNotExists(pinned, image.Id)
+		}
+	}
+
+	return
+}
+
+// isImageUsed compares the specified image info against the list of in-use images to check whether the image ID, tag, or digest is in the in-use list
+func isImageUsed(inUse []string, image imageMgmtImageInfo) bool {
+	if lo.Contains(inUse, image.Id) {
+		return true
+	}
+
+	for _, ref := range image.RepoTags {
+		if lo.Contains(inUse, ref) {
+			return true
+		}
+	}
+
+	for _, ref := range image.RepoDigests {
+		if lo.Contains(inUse, ref) {
+			return true
+		}
+	}
+
+	for _, ref := range image.Names {
+		if lo.Contains(inUse, ref) {
+			return true
+		}
+	}
+
+	return false
+}
+
+// GetRemovalCandidates gets the list of images sorted by creation timestamp, filtering out in-use and pinned images
+func (c *ImageMgmtClient) GetRemovalCandidates() (removalCandidates []imageMgmtImageInfo, rc error) {
+	inUse, err := c.GetInuseImages()
+	if err != nil {
+		rc = fmt.Errorf("failure getting list of in-use images: %w", err)
+		return
+	}
+
+	pinned, err := c.GetPinnedImages()
+	if err != nil {
+		rc = fmt.Errorf("failure getting list of pinned images: %w", err)
+		return
+	}
+
+	// Get a list of candidate images for removal, excluding pinned and in-use images
+
+	var images []imageMgmtImageInfo
+
+	output, err := c.Executor.Execute("podman", "images", "--format", "json")
+	if err != nil {
+		rc = fmt.Errorf("failed to run podman images command: %w", err)
+		return
+	}
+
+	if err := json.Unmarshal([]byte(output), &images); err != nil {
+		rc = fmt.Errorf("unable to parse podman images command output: %w", err)
+		return
+	}
+
+	// Sort images by age, with dangling images first
+	sort.SliceStable(images, func(i, j int) bool {
+		// If Dangling bool matches, sort by Created
+		if images[i].Dangling == images[j].Dangling {
+			return images[i].Created < images[j].Created
+		}
+		// Dangling images first
+		return images[i].Dangling
+	})
+
+	// Filter list of images to get candidates for removal
+	for _, image := range images {
+		// Skip in-use or pinned images
+		if lo.Contains(pinned, image.Id) || isImageUsed(inUse, image) {
+			continue
+		}
+
+		removalCandidates = append(removalCandidates, image)
+	}
+
+	return
+}
+
+// CleanupUnusedImages iterates through the image removal candidates,
+// deleting in sets of 5 until the container storage disk usage threshold is met
+func (c *ImageMgmtClient) CleanupUnusedImages(thresholdPercent int) error {
```

Comment:
I wonder not easier to set threshold to 70% and just run 
`crictl rmi --prune` ? I mean that in 10 minutes those images will be not relevant at all as this node will be rebooted in new ocp and this will require much less logic

Reply:
The images would potentially be relevant if a rollback occurs. The goal is to delete as few images automatically as possible, just to get under the threshold.